### PR TITLE
Patch-set for BSD

### PR DESCRIPTION
**Pull request by LongDirtyAnimAlf**
Hi All,

This is a huge patch-set for BSD (and also a bit of Linux and Android).
OpenBSD was a challenge due to memory protection (w^x).
A lot of work ... ;-)

All greens on all BSD systems.
All greens on all Android systems.

The needed binaries can be found here:
https://github.com/LongDirtyAnimAlf/mORMot/tree/master/static/x86_64-freebsd
https://github.com/LongDirtyAnimAlf/mORMot/tree/master/static/x86_64-openbsd
https://github.com/LongDirtyAnimAlf/mORMot/tree/master/static/aarch64-linux

Please review.

### CODE REVIEW

**Comment by LongDirtyAnimAlf** [comment]
Please skip this pull.
Its not good enough yet.

File: SynCommons.pas [hunk 1]
```diff
@@ -35100,10 +35109,10 @@ function StrCompSSE42(Str1, Str2: pointer): PtrInt;
 
 function crc32csse42(crc: cardinal; buf: PAnsiChar; len: cardinal): cardinal;
 {$ifdef FPC}nostackframe; assembler; asm {$else}
-asm .noframe // ecx=crc, rdx=buf, r8=len (Linux: edi,rsi,rdx)
+asm .noframe // ecx=crc, rdx=buf, r8=len (Linux: edi,rsi,edx)
 {$endif FPC}
         mov     eax, crc
-        not     eax
```
Comment (synopse) [inline SynCommons.pas]: I don't get the difference, but "xor eax,-1" is slower since longer

File: SynCrtSock.pas [hunk 2]
```diff
@@ -4881,18 +4885,33 @@ procedure TCrtSocket.SetInt32OptionByIndex(OptName, OptVal: integer);
...
       fPort := aPort;
     fServer := aServer;
-    fSock := CallServer(aServer,Port,doBind,aLayer,Timeout); // OPEN or BIND
+    while (aRetryCount<RETRYCOUNT) do
+    begin
+      fSock := CallServer(aServer,Port,doBind,aLayer,Timeout); // OPEN or BIND
+      if fSock>0 then break;
+      if (NOT WSAIsFatalError) then
+      begin
+        sleep(RETRYSLEEPTIME);
```
Comment (synopse) [inline SynCrtSock.pas]: please make a comment to specify why waiting is required here (perhaps it should be the caller, or at least been an option, not always retry)

**Comment by LongDirtyAnimAlf** [comment]
Sorry.
I have withdrawn the whole commit.
It was not yet good enough.

Ps:
the above was necessary on FreeBSD and OpenBSD ... perhaps my VM's are too slow.

**Pull request closed without merging**